### DISCUSSION TITLE
Piston Movement

### DISCUSSION THREAD

**Discussion by RhinoCD** [GriefPrevention/GriefPrevention#1924] [Q&A]
What is the difference between Everywhere and Everywhere Simple, I tried reading explanations, and even on reddit, and still don't quite understand. Which one should I choose and why, and why not choose the other?

**Comment by Jikoo** [comment]
Could you explain what you don't understand about the documentation? That would help us improve it. https://docs.griefprevention.com/configuration/#piston-movement

**Reply by RhinoCD** [reply]
I just don’t understand the difference, whats the impact between everything vs the everything simple. Is it a performance thing? which one is better to choose? Why are there 2 different options?

**Reply by RhinoCD** [reply]
I can’t really say the documentation is flawed in any way without knowing what the option does. Maybe a video or visual view explains what’s going on would help, and what impact performance wise it had on the server for either options.

**Reply by Jikoo** [reply]
Yes, it's a performance thing.

`EVERYWHERE` checks every single block individually. `EVERYWHERE_SIMPLE` checks a rectangular box around all affected blocks.

From the writeup for `EVERYWHERE`:
> All blocks moved are checked to ensure that they do not enter into a claim that is not owned by the owner of the piston’s claim.

From the writeup for `EVERYWHERE_SIMPLE`:
> Instead of checking every block moved, the zone surrounding all moved blocks is checked. As long as no part of the zone overlaps a claim with a different owner, pistons are allowed to move freely. This means that certain non-threatening movements will be blocked, but it handles more complex structures much faster.
